conda_build/metadata.py

```diff
@@ -1818,11 +1868,11 @@ def _get_used_vars_meta_yaml(self, force_top_level=False):
         if is_output and not force_top_level:
             recipe_text = self.extract_single_output_text(self.name())
         else:
-            recipe_text = self.get_recipe_text(force_top_level=force_top_level).replace(
-                self.extract_outputs_text().strip(), '')
-
-        recipe_text_without_requirements = recipe_text.replace(
-            self.extract_requirements_text(force_top_level=force_top_level).strip(), "")
+            recipe_text = (self.get_recipe_text(force_top_level=force_top_level).replace(
+                                self.extract_outputs_text().strip(), '') +
+                           self.extract_single_output_text(self.name()))
+        reqs_re = re.compile(r"requirements:.+?(?=^\w|\Z|^\s+-\s(?:name|type))", flags=re.M | re.S)
```

Comment:
This is a more general way of clearing requirements whether they are top-level or output section requirements